modules/harvest/tests/src/Functional/HarvestCodeConsolidationTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\datastore\Functional;
+
+use Drupal\FunctionalTests\Update\UpdatePathTestBase;
+use Drupal\harvest\HarvestService;
+use Drupal\sample_content\SampleContentService;
+use PHPUnit\Exception;
+
+/**
+ * Tests update functions for the datastore module.
+ *
+ * @group datastore
+ * @group update
+ */
+class HarvestCodeConsolidationTest extends UpdatePathTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setDatabaseDumpFiles() {
+    $this->databaseDumpFiles = [
+      __DIR__ . '/../../../../../tests/fixtures/update/update-2.20.0.php.gz',
+    ];
+  }
+
+  /**
+   * Test datastore module update 10001.
+   */
+  public function testUpdates10001on(): void {
+    // HarvestRun() should fail before update because harvest library
+    // namespaces are included in class names stored in the database.
+    /** @var HarvestService $harvest_service */
+    $harvest_service = \Drupal::service('dkan.harvest.service');
+    try {
+      $harvest_service->runHarvest('sample_content');
+    }
+    catch (\Exception $exception) {
+      $exception_msg = $exception->getMessage();
+    }
+    $this->assertSame('No items found to extract, review your harvest plan.', $exception_msg);
+
+    // Run update and recreate harvest service with updated data.
+    $this->runUpdates();
+    $harvest_service = \Drupal::service('dkan.harvest.service');
+
+    try {
+      $result = $harvest_service->runHarvest('sample_content');
+    }
+    catch (\Exception $exception) {
+    }
```

Comment:
I must be missing something - there is no reason for a try/catch here, because no exception is expected, right?
```suggestion
    $result = $harvest_service->runHarvest('sample_content');
```

Reply:
We don't expect an exception, but we also don't expect the test to fail. If there's no exception the test will almost certainly pass. 

The only value to the catch block, is that in the case of failure, it will fail on a failed assertion (` $this->assertIsArray($result);`) instead of on an exception.

Reply:
Let's please remove the try/catch. We should never have a try/catch in a test unless we definitely expect an exception, as above. Having the test throw an error rather than a failure if an unexpected exception is thrown is the correct behavior.

Reply:
Thanks! I'm pretty new to writing unit tests, still learning the conventions.